src/DSP/AnalogFilter.cpp

```diff
@@ -391,7 +391,11 @@ void AnalogFilter::filterout(float *smp)
 
     if(needsinterpolation) {
         //Merge Filter at old coeff with new coeff
+#ifdef _MSC_VER
+        const auto ismp = static_cast<float*>(_alloca(buffersize * sizeof(float)));
+#else
         float ismp[buffersize];
+#endif
```

Comment:
Couldn't you just do something like `auto ismp = new float[buffersize];` for both GCC and MSVC? I am pretty much a novice at C++ so I would be interested to see why something like this wouldn't work. Are we on C++11?

Reply:
That would work from a C++ perspective, but this is realtime code, so heap allocations (using `new`, `make_unique`, etc.) aren't appropriate. There's no bound on how long the memory allocation will take, so you can't guarantee the audio will be generated in time.

Standard C++ doesn't support variable-length stack arrays, which is why this is done in different ways for different compilers. GCC supports such arrays as a non-standard language extension, and MSVC supports the non-standard function `_alloca` to do the same thing. (GCC also supports `alloca` from what I understand, but I'm following the technique we've used in the past in LMMS for compatibility.) Ideally we would stick to standard C++, which would probably involve doing a heap allocation ahead of time on a non-realtime thread. But I'm trying to change as little as possible in this PR as this repo mostly contains third-party code.

Reply:
Ahh I see, thank you for explaining! I hope to help contribute to getting the MSVC build working when I get a chance so this is good to know!